### PR TITLE
Allow specifying AWS AssignPublicIP field

### LINKED ISSUE #4149: [AWS] Allow to set `assignPublicIP` during NodeDeployment creation
**User Story**
As a user of Kubermatic i might have a cluster which is running in a private subnet.
Thus, my nodes don't need a public IP.

If a AWS Machine should not get a public IP, `assignPublicIP` must be set to `false`: https://github.com/kubermatic/machine-controller/blob/master/examples/aws-machinedeployment.yaml#L64

**Acceptance criteria**
- [ ] Its possible to set `assignPublicIP` via the API upon Node/NodeDeployment creation

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4149

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Allowed specifying AWS AssignPublicIP field.
```

### CODE REVIEW

File: api/cmd/kubermatic-api/swagger.json [hunk 1]
```diff
@@ -4844,6 +4844,11 @@
           "type": "string",
           "x-go-name": "AMI"
         },
+        "assignPublicIP": {
+          "description": "should the machine have a publicly accessible IP address",
```
Comment (thz) [inline api/cmd/kubermatic-api/swagger.json]: I think we can be more clear here about what is happening:
This flag controls a property of the AWS instance. When set the AWS instance will get a public IP address assigned during launch overriding a possible setting in the used AWS subnet.

**Review by thz — CHANGES_REQUESTED** [review]

File: api/cmd/kubermatic-api/swagger.json (hunk 1 above)
Comment (maciaszczykm) [reply api/cmd/kubermatic-api/swagger.json]: Okay, that sounds much better.

**Comment by maciaszczykm** [comment]
@thz PTAL

**Review by thz — APPROVED** [review]

**Pull request merged**